src/pfe/portal/modules/FileWatcher.js

```diff
@@ -429,7 +429,7 @@ module.exports = class FileWatcher {
       const projectID = fwProject.projectID;
       const project = this.user.projectList.retrieveProject(projectID);
 
-      if (fwProject.ignoredPaths || event == "newProjectAdded") {
+      if (fwProject.ignoredPaths || fwProject.hasOwnProperty('refPaths') || event == "newProjectAdded") {
         // Send all file watcher clients project related data when a new project is added or ignored paths has changed
         const ignoredPaths = fwProject.ignoredPaths;
```

Comment:
you cannot append the check here. 
if the event is settingsChanged for `refPaths`, the `ignoredPaths` would be undefined. 
And the update would overwrite the `ignoredPaths` in filewatcherd record and project.inf file.

Reply:
The change here: https://github.com/eclipse/codewind/pull/1535/files#diff-e992ab6a23b4524d0a71d4ee49d71101R116 is meant to ensure that whenever `.cw-settings` or `.cw-refpaths.json` file changes, we would invoke the `projectSpecification` handler with both `ignorePaths` and `refPaths` data, so the `projectSettingsChanged` event emitted back to portal would contain both pieces of data.

Reply:
Rethink of this one:
The `projectSettingsChanged` does not work like that. Each setting is an independent setting, which means, if multiple setting has been changed, it would emit multiple `projectSettingsChanged` event. And it would only emit if the setting has been changed. 

Even with the change you made https://github.com/eclipse/codewind/pull/1535/files#diff-e992ab6a23b4524d0a71d4ee49d71101R116 , this check would overwrite `ignoredPaths`.

Reply:
I think I see what you mean.  Will look into it in the morning.

Reply:
@yangcao77 thanks for catching this.  I made 2 new commits 99f42bc and 8c51dab and tested it.  Now the `projectSettingsEvent` will really have both `ignoredPaths` and `refPaths` whenever either changes, and both will then be passed to filewatcherd.

FYI @jgwest

Reply:
The changes look good to me.